tls/s2n_recv.c

```diff
@@ -128,6 +129,9 @@ ssize_t s2n_recv(struct s2n_connection *conn, void *buf, ssize_t size, s2n_block
         if (r < 0) {
```

Comment:
Another thing: Should we expire the entry on an incoming alert? https://github.com/awslabs/s2n/blob/master/tls/s2n_recv.c#L151